python/pyarrow/_dataset_parquet.pyx

```diff
@@ -209,6 +209,26 @@ cdef class ParquetFileFormat(FileFormat):
 
     def make_fragment(self, file, filesystem=None,
                       Expression partition_expression=None, row_groups=None):
+        """
+        Make a FileFragment from a given file.
+
+        Parameters
+        ----------
+        file : file-like object, path-like or str
+            The file or file path to make a fragment from.
+        filesystem : Filesystem, optional
+            If `filesystem` is given, `file` must be a string and specifies
+            the path of the file to read from the filesystem.
+        partition_expression : Expression, optional
+            The filter expression.
+        row_groups : Iterable, optional
```

Comment:
Iterable of what?  I assume numbers (I don't know if specify the iterable type elsewhere).  Also, maybe "The indices of the row groups to include"

Can this be empty?

Reply:
Thanks @westonpace. I'd love to add types https://github.com/apache/arrow/issues/32609 but right now it is just guessing indeed.

I just checked, and an empty list `[]` is allowed, and will not ready any row groups, and return an empty table. Setting it to `None` will return all the row groups.